### PR TITLE
Convert external source to entity

### PR DESCRIPTION
**Pull request by Atmire-Kristof**
This PR expands on #539 and implements a way for the user to import external entries into DSpace through the lookup window.

Removed the option to select external entries in the lookup window and replaced it by an "import" button.
Clicking the button opens an import window for the relevant external entry.

This window displays:
- Information about the external entry (display value and uri)
- A text input for the collection ID to import the new entry into (this is a placeholder and will be replaced by a dropdown of sorts). The collection we're submitting in is automatically entered here.
- A selectable list of similar entities already present in the repository
- An option saying "new entity"
- An option saying "new authority"

Selecting any of the options will automatically deselect the other options.

Clicking import with a local entity selected will simply add the local entity to the selected list in the lookup window.
Clicking import with new entity selected will perform an import of the external entry into a DSpace item for the collection entered. After the import is complete, the new entity is added to the selection on the lookup window.
The authority option is a placeholder and currently doesn't do anything when selected.

### CODE REVIEW

**Comment by paulo-graca** [comment]
import_entity_pg_comments

**Comment by Atmire-Kristof** [comment]
Thanks for the comment @paulo-graca ,

the collection input is a temporary placeholder, it will eventually be turned into a proper dropdown.
I have however moved the input below the "entities" label, as it indeed makes more sense to be positioned here.

As mentioned in the PR description, the authority option is also a placeholder. To avoid confusion, I've added the boolean property `authorityEnabled` to the component and set it to false. This hides the authority section of the import window for now.

**Comment by benbosman** [comment]
The commits specific to this branch (excluding #539) can be seen at https://github.com/atmire/dspace-angular/pull/4 to ease the code review

**Comment by paulo-graca** [comment]
Just to note that I'm getting this error when I open lookup modal window:

```
ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'ngIf: false'. Current value: 'ngIf: true'.
    at viewDebugError (client.js:63089)
    at expressionChangedAfterItHasBeenCheckedError (client.js:63077)
    at checkBindingNoChanges (client.js:63179)
    at checkNoChangesNodeInline (client.js:66050)
    at checkNoChangesNode (client.js:66039)
    at debugCheckNoChangesNode (client.js:66643)
    at debugCheckDirectivesFn (client.js:66571)
    at Object.eval [as updateDirectives] (AppComponent.html:13)
    at Object.debugUpdateDirectives [as updateDirectives] (client.js:66560)
    at checkNoChangesView (client.js:65938)
```

**Comment by tdonohue** [comment]
PR #539 was just merged. So, this PR likely needs a quick rebase (since it was dependent on 539)

**Comment by Atmire-Kristof** [comment]
@paulo-graca Thank you for noticing this. This issue was also present on the latest master branch and originates from how the loading property in AppComponent is updated in `ngAfterViewInit`. Adding a `delay(0)` operator seems to fix the issue. Detailed explanation to why this fixes the error can be found in this article: https://blog.angular-university.io/angular-debugging/
Testing after applying the fix seemed to have fixed the issue (at least on my end).

@tdonohue I've merged the latest master into this PR.

File: src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/dynamic-lookup-relation-modal.component.html [hunk 1]
```diff
@@ -25,12 +25,14 @@ <h4 class="modal-title" id="modal-title">{{ ('submission.sections.describe.relat
                  [title]="'submission.sections.describe.relationship-lookup.search-tab.tab-title.' + source.id | translate  : {count: (totalExternal$ | async)[idx]}">
```
Comment (paulo-graca) [inline src/app/shared/form/builder/ds-dynamic-form-ui/relation-lookup-modal/dynamic-lookup-relation-modal.component.html]: I'm getting this error:

```
DsDynamicLookupRelationModalComponent.html:24 ERROR Error: ExpressionChangedAfterItHasBeenCheckedError: Expression has changed after it was checked. Previous value: 'title: [object Object]'. Current value: 'title: Sherpa Journals (0)'.
    at viewDebugError (client.js:63289)
    at expressionChangedAfterItHasBeenCheckedError (client.js:63277)
    at checkBindingNoChanges (client.js:63379)
    at checkNoChangesNodeInline (client.js:66250)
    at checkNoChangesNode (client.js:66239)
    at debugCheckNoChangesNode (client.js:66843)
    at debugCheckDirectivesFn (client.js:66771)
    at Object.eval [as updateDirectives] (DsDynamicLookupRelationModalComponent.html:24)
    at Object.debugUpdateDirectives [as updateDirectives] (client.js:66760)
    at checkNoChangesView (client.js:66138)
View_DsDynamicLookupRelationModalComponent_2 @ DsDynamicLookupRelationModalComponent.html:24
./node_modules/@angular/core/fesm5/core.js.DebugContext_.logError @ client.js:66988
./node_modules/@angular/core/fesm5/core.js.ErrorHandler.handleError @ client.js:58621
(anonymous) @ client.js:60975
./node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ client.js:203905
./node_modules/zone.js/dist/zone.js.Zone.run @ client.js:203664
./node_modules/@angular/core/fesm5/core.js.NgZone.runOutsideAngular @ client.js:60107
./node_modules/@angular/core/fesm5/core.js.ApplicationRef.tick @ client.js:60975
(anonymous) @ client.js:60859
./node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ client.js:203905
onInvoke @ client.js:60148
./node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke @ client.js:203904
./node_modules/zone.js/dist/zone.js.Zone.run @ client.js:203664
./node_modules/@angular/core/fesm5/core.js.NgZone.run @ client.js:60062
next @ client.js:60859
schedulerFn @ client.js:56364
./node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.__tryOrUnsub @ client.js:183508
./node_modules/rxjs/_esm5/internal/Subscriber.js.SafeSubscriber.next @ client.js:183446
./node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber._next @ client.js:183390
./node_modules/rxjs/_esm5/internal/Subscriber.js.Subscriber.next @ client.js:183367
./node_modules/rxjs/_esm5/internal/Subject.js.Subject.next @ client.js:183133
./node_modules/@angular/core/fesm5/core.js.EventEmitter.emit @ client.js:56348
checkStable @ client.js:60117
onLeave @ client.js:60184
onInvokeTask @ client.js:60142
./node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask @ client.js:203936
./node_modules/zone.js/dist/zone.js.Zone.runTask @ client.js:203709
./node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask @ client.js:204012
invokeTask @ client.js:205258
globalZoneAwareCallback @ client.js:205284
DsDynamicLookupRelationModalComponent.html:24 ERROR CONTEXT DebugContext_ {view: {…}, nodeIndex: 1, nodeDef: {…}, elDef: {…}, elView: {…}}
```

**Comment by artlowel** [comment]
>I've tested this functionality and noticed when importing an author, the changes to the current submission item are not always visible.
I can see the calls to create the item and the relationship, but the updates to the workspace item are not retrieved automatically. The updates are retrieved with a delay, and are only displayed after I performed a few other actions. This may indicate the cache is not invalidated after the relationship has been created.

@benbosman That is a known issue with #531, that will be addressed in #541
It isn't caused by the changes this PR.

**Pull request merged**